### ISSUE TITLE
private_bounds: false positive on inherent impl with no public items

### ISSUE THREAD

**Issue by Kyuuhachi** [rust-lang/rust#139158]
### Code

```Rust
#![allow(unused)]
struct Type<T>(T);
mod a {
    trait Trait {}
    impl<T: Trait> super::Type<T> {}
}
```

### Current output

```Shell
warning: trait `Trait` is more private than the item `a::<impl Type<T>>`
 --> src/lib.rs:5:5
  |
5 |     impl<T: Trait> super::Type<T> {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ implementation `a::<impl Type<T>>` is reachable at visibility `pub(crate)`
  |
note: but trait `Trait` is only usable at visibility `pub(self)`
 --> src/lib.rs:4:5
  |
4 |     trait Trait {}
  |     ^^^^^^^^^^^
  = note: `#[warn(private_bounds)]` on by default
```

### Desired output

```Shell
No warning
```

### Rationale and extra context

No items inside the impl exist that leak private details, and the mere existence of the impl shouldn't be an issue. Same warning appears if the impl is non-empty.

I think the warning should be on individual items inside the impl rather than the impl itself.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.88.0-nightly (1799887bb 2025-03-29)
binary: rustc
commit-hash: 1799887bb281d1ab49287750f1950b8c738c6b77
commit-date: 2025-03-29
host: x86_64-unknown-linux-gnu
release: 1.88.0-nightly
LLVM version: 20.1.1
```

**Comment by Urgau** [comment]
The lint is defined as[^1]:

> The `private_bounds` lint detects types in a secondary interface of an item, that are more private than the item itself. Secondary interface of an item consists of bounds on generic parameters and where clauses, including supertraits for trait items.

and by that definition, `Trait` is indeed more private than `Type`. If you made `trait Trait`: `pub(super)` then they would be at the same level and the lint wouldn't trigger[^2].

> I think the warning should be on individual items inside the impl rather than the impl itself.

Can you say more/elaborate on this part? As far as I can tell the lint already note the item that is more private.

[^1]: https://doc.rust-lang.org/beta/rustc/lints/listing/warn-by-default.html#private-bounds
[^2]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=b042464a7217d2f18fe417256058acce

**Comment by Manishearth** [comment]
> Can you say more/elaborate on this part? As far as I can tell the lint already note the item that is more private.

The warning says "implementation `a::<impl Type<T>>` is reachable...".

But impls aren't "reachable". Impls aren't nameable at all. The impl in this issue is _empty_, which is definitely not reachable, but you get similar issues if you add private methods.
